conf.d/forgit.plugin.fish

```diff
@@ -39,7 +39,8 @@ function forgit::log -d "git commit viewer"
     forgit::inside_work_tree || return 1
 
     set files (echo $argv | sed -nE 's/.* -- (.*)/\1/p')
-    set preview "echo {} |grep -Eo '[a-f0-9]+' |head -1 |xargs -I% git show --color=always % -- $files | $forgit_show_pager"
+    set preview_cmd "echo {} |grep -Eo '[a-f0-9]+' |head -1 |xargs -I% git show --stat --color=always % -- $files | $forgit_show_pager"
+    set diff_cmd 'echo {} | grep -Eo "[a-f0-9]+" | head -1 | xargs -I % bash -ic "forgit::diff %^1 %"'
```

Comment:
Could you extract the shared portion of these into a shared variable just to reduce duplication

```
echo {} |grep -Eo '[a-f0-9]+' |head -1
```

Reply:
Good idea! Will do that next week, I am on holidays for the weekend. 🙂

Reply:
nice! have a good trip :)

Reply:
@cjappl I have implemented your requested change in a separate commit, because it affects a lot more code than the sections I had modifed in the first place. I find it really useful for the code consistency, though, so I added it directly to this PR.
There was also a tiny bug left for git::cherry_pick, where the SHA was not correctly parsed if `FORGIT_LOG_GRAPH_ENABLE` was set to `false`. I fixed this on the way, since it was the only command that parsed the SHA differently than the other commands.
I tested the changes roughly in zsh and fish and rebased the PR on the current master, so we should be ready to go.

Reply:
Wow!! Big improvement, appreciate your thoroughness. 

Going to assign wfxr so he can give final approval